### PR TITLE
Fix Android issues

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

- Adds a method to platform channel to trigger the user cancellation exception on Android.
- The above fix also fixes the race condition caused due to login method
- Update README to show how to use `setResultUrl` method

### CODE REVIEW

**Review by grvgoel81 — APPROVED** [review]
lgtm.

File: example/lib/main.dart [hunk 1]
```diff
@@ -20,14 +20,29 @@ class MyApp extends StatefulWidget {
...
+  void dispose() {
+    super.dispose();
+    WidgetsBinding.instance.removeObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(final AppLifecycleState state) {
+    // This is important to trigger the user cancellation on Android.
+    if (state == AppLifecycleState.resumed) {
+      Web3AuthFlutter.setResultUrl();
```
Comment (chaitanyapotti) [inline example/lib/main.dart:44]: can you add platform identifiers for this? or add setResultUrl to do nothing in case of ios?

Comment (AyushBherwani1998) [reply example/lib/main.dart:44]: The setResultUrl handles it internally so user doesn't have to worry.

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**